### PR TITLE
ddl: support table placement option

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: particial close https://github.com/pingcap/tidb/issues/26581

### What is changed and how it works?

1: Support create table with placement options, which could be direct placement options or placement policy.
When both this two exists, placement policy take the priority to take effect in the schedule of PD.

2: Since we move the policyInfo into the parser, so we change the reference source to parser.

### Check List 

Tests 

- [x] Unit test
- [x] Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1609,6 +1608,21 @@ func checkTableInfoValidWithStmt(ctx sessionctx.Context, tbInfo *model.TableInfo
 			}
 		}
 	}
+	if tbInfo.DirectPlacementOpts != nil {
```
Comment (xhebox) [inline ddl/ddl_api.go:1611]: What if `DirectPlacementOpts  == nil` and `tbInfo.PlacementPolicyRef != nil`? Do we need to assign `PlacementPolicyRef.ID`?

File: ddl/placement_policy_test.go [hunk 2]
```diff
@@ -194,3 +192,95 @@ func (s *testDBSuite6) TestConstraintCompatibility(c *C) {
...
+	c.Assert(err.Error(), Equals, "[schema:8239]Unknown placement policy 'x'")
+	tk.MustExec("drop table if exists t")
+
+	tk.MustExec("create placement policy x " +
+		"PRIMARY_REGION=\"cn-east-1\" " +
+		"REGIONS=\"cn-east-1, cn-east-2\" " +
+		"FOLLOWERS=2 " +
+		"FOLLOWER_CONSTRAINTS=\"[+zone=cn-east-1]\" " +
+		"CONSTRAINTS=\"[+disk=ssd]\" ")
+	// The direct placement option is conflict with the named policy.
```
Comment (xhebox) [inline ddl/placement_policy_test.go:255]: The case of named policy only?

**Review by xhebox — COMMENTED** [review]
I just noticed that in the RFC:

> You can not use both a placement policy and direct assignment. If you alter specify both in a CREATE TABLE or ALTER TABLE an error will be returned.

Maybe we should not override?

File: ddl/ddl_api.go (hunk 1 above)
Comment (AilinKid) [reply ddl/ddl_api.go:1615]: good catch~ addressed.

**Review by xhebox — COMMENTED** [review]
Looks like format is still broken.

**Review by xhebox — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**